### ISSUE TITLE
Android 12: No manual or automatic update

### ISSUE THREAD

**Issue by joschiv** [corona-warn-app/cwa-app-android#3295]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: Google Pixel 4 XL
- Android version: 12 Beta 1
- App version:2.2.1
- Build Nummer: SPB1.210331.013

## Describe the bug

Button für die Manuelle Aktualisierung wird nicht angezeigt. Nach 4, 6, 8 ,12 oder 24 Stunden wird die Aktualisierung auch nicht gestartet. Akku Optimierung für die Corona Warn App ist abgeschaltet. 

Ich möchte das hier melden da ich bis jetzt noch keine Infos Über die Corona Warn App und Android 12 gefunden habe, das es zu Probleme kommen kann. 

![Screenshot_20210527-131517](https://user-images.githubusercontent.com/56675401/119816989-b2ba1c00-beed-11eb-834c-87c1d3c662b9.png)

**Comment by LeQuadratico** [comment]
I don´t really understand the problem. I think there never was a button for a manual refresh/update. Also, it seems like your status refreshed at 09:07 today, so it should work.
Can you point out what button you mean on one of the official screenshots? https://www.coronawarn.app/de/screenshots/

**Comment by Ein-Tim** [comment]
@LeQuadratico 

Deactivated background app reffest should show a button like this: 
![image](https://user-images.githubusercontent.com/67682506/119818473-8b644e80-beef-11eb-8f47-98927b85bbb1.png)

@joschiv Are you connected via WiFi or via Mobile Data?

**Comment by joschiv** [comment]
It won`t Update at all. This was after reset of the App. So if you disable auto update you will get a button for manual update. It show`s on button with an counter "update in 03:59:12". If you enable battery optimization for the Corona App, this button will show up.
In general after 4 hours it will update. But on Android 12 it only updates after reset of the app or fresh install of the app.

![Screenshot_20210527-131517](https://user-images.githubusercontent.com/56675401/119818467-88695e00-beef-11eb-827c-9b97e8259574.png)

**Comment by MikeMcC399** [comment]
@Ein-Tim 
It looks like your screen shot in https://github.com/corona-warn-app/cwa-app-android/issues/3295#issuecomment-849556155 is from iOS, not Android.

**Comment by Ein-Tim** [comment]
@MikeMcC399 Yes, but the implementation is the same under Android, isn't it?

**Comment by joschiv** [comment]
@Ein-Tim: Both Wifi and 4G.

@MikeMcC399: It`s on Android too.

**Comment by LeQuadratico** [comment]
Ok, im on the Android 12 Beta 1 as well. To test it I enabled background optimizations. After restarting my phone (Pixel 3a) I can confirm that there is no button showing up.

**Comment by Ein-Tim** [comment]
@joschiv 

Could you turn off 4G and open the app again? If the app detects that you're connected via Mobile Data it will only update all 24h.

**Comment by MikeMcC399** [comment]
@joschiv 
CWA Android 2.2.1 is built against Android 10 (API 29)
https://github.com/corona-warn-app/cwa-app-android/blob/595ab630dfad5ac1e29021c084c400054193a498/Corona-Warn-App/build.gradle#L28

which is one version less than Android 11.

I suggest adding a wishlist item to https://github.com/corona-warn-app/cwa-wishlist to request compatibility with Android 12 beta.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

Should we maybe add this incompatibility information to the FAQ (similar to the entry we had when iOS 14 was in Beta)?

**Comment by joschiv** [comment]
@Ein-Tim it doesnt change anything. It wont`t update after 24h, 48h or 72h. 

@MikeMcC399 I know this. So this "bug" i want to tell everybody there is a problem on Android 12 Beta so don`t update if you want to use the Corona Warn App. I know this is a Beta and in general there is no support. This is fine.

**Comment by LeQuadratico** [comment]
It seems like it is working when you disable background optimizations, so this could be a workaround for Beta-users.

**Comment by joschiv** [comment]
> It seems like it is working when you disable background optimizations, so this could be a workaround for Beta-users.

Not for me on this device. After disable background optimizations and restart, it`s the same. Not updates after 4h, 12h, or 72 hours.

**Comment by LeQuadratico** [comment]
Ok thats strange. Mine updates automatically multiple times a day as in Android 11.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
I have never seen "Aktualisierung" on Android, so I assume that it only appears on iOS. I am running Android 8, 10 and 11. Since this is a specific Android issue, I suggest to keep iOS screenshots out of this issue.

![image](https://user-images.githubusercontent.com/66998419/119820948-4f7eb880-bef2-11eb-9ed6-c93cb7e378fd.png)

**Comment by joschiv** [comment]
> @Ein-Tim
> I have never seen "Aktualisierung" on Android, so I assume that it only appears on iOS. I am running Android 8, 10 and 11. Since this is a specific Android issue, I suggest to keep iOS screenshots out of this issue.
> 
> ![image](https://user-images.githubusercontent.com/66998419/119820948-4f7eb880-bef2-11eb-9ed6-c93cb7e378fd.png)

You get the button after you enable battery optimizations. i had many Android 10 and 11 devices like Samsung Note 10, 20 Ultra, S20, Pixel 3 XL, Pixel 4 XL. This button will show up. 100%

But this is not the main problem. The main problem on Android 12 is, it dosen`t update at all.

**Comment by MikeMcC399** [comment]
I've added https://github.com/corona-warn-app/cwa-app-android/issues/3297 as an enhancement request to get the app updated to build against native Android 11. That may help with some compatibility issues with Android 12 beta.

**Comment by MikeMcC399** [comment]
@joschiv
> The main problem on Android 12 is, it dosen`t update at all.

I suggest then to concentrate on this issue. Your first screen shot shows that an exposure check ran today at 09:07. So normally if you have a WiFi connection, this should get updated about every 4 hours, and if you have only a data connection it takes 24 hours before there is an update again. If you have neither of these, then after 36 hours you should get a "deadman notification" "Please remember to open the Corona-Warn-App regularly". (In German: "Bitte denken Sie daran, die Corona-Warn-App regelmäßig zu öffnen, um Ihren Risikostatus zu überprüfen.")

Perhaps monitor the situation over the next 24 to 36 hours and see what happens?

**Comment by MikeMcC399** [comment]
@joschiv 
Here are a few thoughts about how you may find out a little more, if you continue to have issues with missing risk assessment:

- You can check in Android Settings > COVID-19 Benachrichtigungen > Drei-Punkte Symbol > Überprüfung auf mögliche Begegnungen, to see if Google has registered any checks.
- In the CWA App > Drei-Punkte Symbol > App-Informationen > Fehlerberichte > STARTEN, then through "TEILEN UND FORTSETZEN" you can capture a log file, and send it for instance to Google Drive where you can look at it.
- If you scroll down on the CWA Startseite to "Bestätigte Neuinfektionen", if the date stamp is "Gestern" then that shows your device is connecting to the server where it also gets the diagnosis keys which it needs to carry out the risk assessment.

Let us know how your device behaves over the next 24 - 36 hours.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
> Should we maybe add this incompatibility information to the FAQ (similar to the entry we had when iOS 14 was in Beta)?

I suggest you take your question / suggestion as a new issue to the https://github.com/corona-warn-app/cwa-website where we can discuss further and track it.

**Comment by joschiv** [comment]
> @joschiv
> Here are a few thoughts about how you may find out a little more, if you continue to have issues with missing risk assessment:
> 
> * You can check in Android Settings > COVID-19 Benachrichtigungen > Drei-Punkte Symbol > Überprüfung auf mögliche Begegnungen, to see if Google has registered any checks.

No checks at all. This was the reason to post it here. 

> * In the CWA App > Drei-Punkte Symbol > App-Informationen > Fehlerberichte > STARTEN, then through "TEILEN UND FORTSETZEN" you can capture a log file, and send it for instance to Google Drive where you can look at it.
> * If you scroll down on the CWA Startseite to "Bestätigte Neuinfektionen", if the date stamp is "Gestern" then that shows your device is connecting to the server where it also gets the diagnosis keys which it needs to carry out the risk assessment.
> 
> Let us know how your device behaves over the next 24 - 36 hours.

After playing around with the device, I tried to re-login with my Google account. Because the Covid 19 settings are in the Google Account settings page on Android 12. 
It was not possible because the settings app crashed. So I reinstalled Android 12 on my device again. 
After setting up everything i installed the cwa. And now after 6 hours I got my first update. 
And now I can  see checks in the settings page. 

So far everything is running.

**Comment by MikeMcC399** [comment]
@joschiv 
> So far everything is running.

It's good to hear that exposure logging and risk assessment are working for you after you reinstalled your Android 12 beta.

In parallel yesterday I installed CWA 2.2.1 in a Pixel 3a emulator with API 31 (S) (Build Nummer: SPB1.210331.013.A1) and it is also doing exposure checks. @LeQuadratico also said in https://github.com/corona-warn-app/cwa-app-android/issues/3295#issuecomment-849566975 that there was no problem with their device.

So it looks like there is no general incompatibility with the Android 12 beta and probably this issue can be closed.

**Comment by heinezen** [comment]
@joschiv Thanks for the update! Good that it's working as intended again.

I'll take @MikeMcC399 's suggestion and close the issue. We'll reopen if the problem occurs again for someone.

----
Corona-Warn-App Open Source Team
